sql_grader/mixins/grading.py

```diff
@@ -1,21 +1,74 @@
 """
 Mixin grading-related functionality
 """
+import logging
+import os.path
+
+from django.utils.translation import ugettext_lazy as _
+
+from codejail.safe_exec import safe_exec, SafeExecException
+
+from xblock.fields import Boolean
 from xblock.fields import Float
 from xblock.fields import Integer
 from xblock.fields import Scope
+from xblock.fields import String
+
 from xblock.scorable import ScorableXBlockMixin
 from xblock.scorable import Score
 
-from ..models import SqlProblem
-from ..models import DATABASES
+from ..problem import all_datasets
 
 
-def _(text):
+log = logging.getLogger('sql_grader')
+
+
+# pylint: disable=unused-argument
+def attempt_safe(dataset, answer_query, verify_query, is_ordered, query):
     """
-    Mock translation for scraping
+    Attempt a SqlProblem, using codejail to sandbox the execution.
     """
-    return text
+    results = {
+        'answer_query': answer_query,
+        'dataset': dataset,
+        'verify_query': verify_query,
+        'is_ordered': is_ordered,
+        'query': query
+    }
+    code = """
+from sql_grader.problem import SqlProblem
+submission_result, answer_result, error, comparison = SqlProblem(
+    answer_query=answer_query,
+    dataset=dataset,
+    verify_query=verify_query,
+    is_ordered=is_ordered
+).attempt(query)
+
+"""
+    python_path = [os.path.abspath(
+        os.path.join(
+            os.path.dirname(__file__),
+            '../..'
+        )
+    )]
+
+    try:
+        safe_exec(code, results, python_path=python_path, slug='sql_grader')
+    except SafeExecException:
+        log.exception(query)
+        # how should resource limits be communicated to the user?
+        results = {
+            'submission_result': None,
+            'answer_result': None,
+            'error': "Could not execute query",
```

Comment:
Is this the message displayed to the user? If so, could we flesh it out just a bit more?
Something like:

> We could not execute your query; please try again.

Also, if user-facing, we should add `i18n`.